### ISSUE TITLE
Slot is not working in rare and unpredictable cases

### ISSUE THREAD

**Issue by Instinctlol** [vuejs/vue#12215]
### Version
2.6.14

### Reproduction link
[https://stackblitz.com/edit/vuejs-starter-uzpckr?file=app.js](https://stackblitz.com/edit/vuejs-starter-uzpckr?file=app.js)

### Steps to reproduce
Please look at the field "additionalArguments", it contains a link.
Press ALT M to simulate fetching new data, instead of rendering a link here, the default slot content for that named slot is rendered instead.
You can press ALT J to load the original data, this time, there's also no link.

### What is expected?
Instead of rendering the default slot content defined by the child for its slot named "arrayValue", the passed down slot from orderDataCard should be rendered instead (the link).

### What is actually happening?
There are 3 components.
The grandchild, which offers the slot named "arrayValue".
The child, which passes the slot between grandchild and parent.
The parent, who decides to render a link, but no link is being shown when data is reloaded.

Instead, the default slot content of the child is being rendered.

---
Grandchild = headlessTable
Child = collapsableCard
Parent = orderDataCard

I noticed that if you uncomment line 134 in app.js, which is the field called 'mark', the issue seems to be resolved.
But in production, which uses a lot more fields than in the example, another field seems to be responsible for this: 'invoices'.

In any case, there is no dependence or anything between any of the fields, so it's hard for me to understand why this happens and I suspect this to be a bug with vue.

I created an issue for this last week https://github.com/vuejs/vue/issues/12206, which has been closed but has not been reopened when modified, so this is why create a new one.

**Comment by posva** [comment]
Your reproduction is still not minimal and very hard to follow with a lot of unnecessary big objects.

---

Remember to use the [forum](http://forum.vuejs.org/) or the [Discord chat](https://vue-land.js.org/) to ask questions!

**Comment by Instinctlol** [comment]
There is only 1 object that contains 4 fields, which are computed into arrays. Please tell me what is bothering you?

**Comment by Instinctlol** [comment]
@posva I deleted a bunch of them. Is it fine for you now?

**Comment by posva** [comment]
No, it's not. A reproduction should be easy to follow, I cannot dive into your code to help you debug the problem. Right now it's still far from being easy to navigate. Give other issues a look, reproductions are rarely more than 100 lines

**Comment by Instinctlol** [comment]
@posva I deleted some more fields and now there's nothing to reduce anymore.

The biggest file is app.js. What you got there is an object that contains 4 fields: mark, common, salesOrganisation, invoices. These are independent from each other and these will be recomputed into arrays in orderDataCard (the "parent") and consequently rendered via child components and one grandchild. The other lines are for another object of the same structure. Obviously there is some sort of dependency or connection for these fields somewhere in vue somehow, else uncommenting the line `// mark: 'm',` wouldn't "solve" the problem.

I also cannot reduce the html code of orderDataCard, because that also fixes the problem. It

So these things are all part of the problem and it's just not possible to minimalize it further. But the code is valid, correct and makes sense. I acknowledge that this is a more complex problem, but just looking at the amount of lines to decide whether the reproduction is minimal is kinda ignorant to be honest.

**Comment by Instinctlol** [comment]
@posva I updated the OP and did everything I could to make it more easy for you. I also did a [stackoverflow](https://stackoverflow.com/questions/68802093/vue-slot-is-not-working-in-rare-and-unpredictable-cases-potential-vue-bug) for this. Again, I can not reduce the code further and I don't think anyone is able to.

Please reopen the issue.

**Comment by yuejunqi123** [comment]
I see

&nbsp;




岳俊奇
前端实习生
天耀宏图


 联系我









&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2021年8月20日(星期五) 下午4:05
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; 
主题: Re: [vuejs/vue] Slot is not working in rare and unpredictable cases (#12215)





 
@posva I updated the OP and did everything I could to make it more easy for you. I also did a stackoverflow for this. Again, I can not reduce the code further and I don't think anyone is able to.
 
Please reopen the issue.
 
—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
